cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDeleteController.java

```diff
@@ -0,0 +1,147 @@
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Delete a pump from a contract",
+        path = "/projects/{office}/{project-id}/water-user/{water-user}/contracts/{contract-id}/pumps/{name}",
+        method = HttpMethod.DELETE,
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(DELETE));
+            String officeId = ctx.pathParam(OFFICE);
+            String projectName = ctx.pathParam(PROJECT_ID);
+            String entityName = ctx.pathParam(WATER_USER);
+            PumpType pumpType = PumpType.valueOf(ctx.queryParam(PUMP_TYPE));
+            String contractName = ctx.pathParam(CONTRACT_NAME);
+            WaterContractDao contractDao = getContractDao(dsl);
+            CwmsId projectLocation = buildCwmsId(officeId, projectName);
+            List<WaterUserContract> contract = contractDao.getAllWaterContracts(projectLocation, entityName);
+
+            for (WaterUserContract waterUserContract : contract) {
+                if (waterUserContract.getContractId().getName().equals(contractName)) {
+                    switch (pumpType) {
+                        case IN:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpInLocation().getPumpLocation().getName(),
+                                    PumpType.IN, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        case OUT:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpOutLocation().getPumpLocation().getName(),
+                                    PumpType.OUT, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        case BELOW:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpOutBelowLocation().getPumpLocation().getName(),
+                                    PumpType.BELOW, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        default:
+                            CdaError error = new CdaError("Invalid pump type provided.");
+                            LOGGER.log(Level.SEVERE, "Invalid pump type provided.");
+                            ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
```

Comment:
This should return an error code 400 as it is bad input by the client. You should be able to throw an IllegalArgumentException here and it'll get handled appropriately.

Reply:
Added a throw statement and removed manual error handling